apps/dashboard/app/javascript/module_browser.js

```diff
@@ -0,0 +1,47 @@
+document.addEventListener('DOMContentLoaded', function () {
+  document.querySelectorAll('[data-name]').forEach(card => {
+    const versions = card.querySelectorAll("[data-role='selectable-version']");
+    const infoBox = card.querySelector("[data-role='module-info']");
+    const dependenciesSpan = infoBox.querySelector("[data-role='module-dependencies']");
+    const loadCmd = infoBox.querySelector("[data-role='module-load-command']");
+
+    const defaultVersion = [...versions].find(v => v.dataset.default === 'true');
+    if (defaultVersion) updateInfo(defaultVersion);
+
+    versions.forEach(badge => {
+      badge.addEventListener('click', () => {
+        versions.forEach(v => v.classList.remove('selected-version'));
+        badge.classList.add('selected-version');
+
+        updateInfo(badge);
+      });
+    });
+
+    function updateInfo(badge) {
+      const module = badge.dataset.module;
+      const version = badge.dataset.version;
+      const deps = badge.dataset.dependencies || '-';
+
+      dependenciesSpan.textContent = deps;
+      loadCmd.textContent = `module load ${module}/${version}`;
+    }
+  });
+
+  document.querySelectorAll('[data-role="copy-btn"]').forEach(button => {
+    button.addEventListener('click', () => {
+      const selector = button.getAttribute('data-clipboard-target');
+      const target = document.querySelector(selector);
+      if (!target) return;
+  
+      const text = target.textContent;
+      navigator.clipboard.writeText(text)
```

Comment:
I'm going to have to look into this - but IIRC this isn't supported by all the browsers yet or there's an ongoing kerfuffle with the standard.

Of course you catch the error below, but I just want to call it out, if nothing else.

Reply:
Well MDN let's us know about compatability - I guess I just need to double check the supported versions if they're recent or what.

https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API